libzcash_script/src/lib.rs

```diff
@@ -385,86 +387,126 @@ mod tests {
-        let ret = check_verify_callback(
-            &CxxInterpreter {
-                sighash: &invalid_sighash,
-                lock_time,
-                is_final,
-            },
-            &RustInterpreter::new(CallbackTransactionSignatureChecker {
-                sighash: &invalid_sighash,
-                lock_time: lock_time.into(),
-                is_final,
-            }),
-            &SCRIPT,
-            flags,
-        );
-
-        assert_eq!(
-            ret.0.map_err(normalize_err),
-            ret.1.clone().map_err(normalize_err)
-        );
-        assert_eq!(ret.1, Ok(false));
+
+        let ret = RustInterpreter::new(interpreter::CallbackTransactionSignatureChecker {
+            sighash: &sighash,
+            lock_time: lock_time.into(),
+            is_final,
+        })
+        .verify_callback(&SCRIPT, flags);
+        assert_eq!(ret, Ok(true));
+    }
+
+    #[test]
+    fn it_fails_with_null_checker() {
+        let flags = interpreter::Flags::P2SH | interpreter::Flags::CHECKLOCKTIMEVERIFY;
+
+        let ret = RustInterpreter::new(interpreter::NullSignatureChecker())
+            .verify_callback(&SCRIPT, flags);
+        assert_eq!(ret, Ok(false));
     }
 
     #[test]
-    fn it_fails_on_missing_sighash() {
+    fn it_fails_on_invalid_sighash_in_cxx() {
         let lock_time: u32 = 2410374;
         let is_final: bool = true;
         let flags = interpreter::Flags::P2SH | interpreter::Flags::CHECKLOCKTIMEVERIFY;
+        let ret = CxxInterpreter {
+            sighash: &invalid_sighash,
+            lock_time,
+            is_final,
+        }
+        .verify_callback(&SCRIPT, flags);
 
-        let ret = check_verify_callback(
-            &CxxInterpreter {
-                sighash: &missing_sighash,
-                lock_time,
-                is_final,
-            },
-            &RustInterpreter::new(CallbackTransactionSignatureChecker {
-                sighash: &missing_sighash,
-                lock_time: lock_time.into(),
-                is_final,
-            }),
-            &SCRIPT,
-            flags,
-        );
-
-        assert_eq!(
-            ret.0.map_err(normalize_err),
-            ret.1.clone().map_err(normalize_err)
-        );
-        assert_eq!(ret.1, Ok(false));
+        assert_eq!(ret, Ok(false));
+    }
+
+    #[cfg(feature = "signature-validation")]
+    #[test]
+    fn it_fails_on_invalid_sighash_in_rust() {
+        let lock_time: u32 = 2410374;
+        let is_final: bool = true;
+        let flags = interpreter::Flags::P2SH | interpreter::Flags::CHECKLOCKTIMEVERIFY;
+        let ret = RustInterpreter::new(interpreter::CallbackTransactionSignatureChecker {
+            sighash: &invalid_sighash,
+            lock_time: lock_time.into(),
+            is_final,
+        })
+        .verify_callback(&SCRIPT, flags);
+
+        assert_eq!(ret, Ok(false));
+    }
+
+    #[test]
+    fn it_fails_on_invalid_sighash_with_null_checker() {
```

Comment:
AFAICT this is identical to `it_fails_with_null_checker` and is not checking anything new. Remove this.